core/kernel/dt.c

```diff
@@ -51,6 +51,101 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
 	return prop;
 }
 
+int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop = NULL;
+	int len_prop = 0;
+	int it_num = -1;
+
+	/*
+	 * Interrupt property is defined with 3x32 bits word:
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 *  - Interrupt Trigger type
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if (!int_prop || (len_prop < 2))
+		return -1;
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+
+	return it_num;
+}
+
+paddr_t dt_node_offset_by_compatible_status(void *fdt, int status, int *node,
+						const char *compatible)
+{
+	paddr_t offset = 0;
+	int fnode = 0;
+	int node_status = 0;
+
+	fnode = fdt_node_offset_by_compatible(fdt, *node, compatible);
+
+	while (fnode != -FDT_ERR_NOTFOUND) {
+		node_status = _fdt_get_status(fdt, fnode);
+		if ((node_status & status) == status) {
```

Comment:
I think here `status` is expected to be an input value, not specifically both a mask and a value.

Reply:
Here the status can be a defined with multiple bits hence it's used as mask first to extract expected status bit and then status is used to check result of the bit mask.

Reply:
Think looks strange. Check which behavior you would expect when calling:
* `dt_node_offset_by_compatible_status(fdt, DT_STATUS_OK_NSEC, &node, &compat)` ?
* `dt_node_offset_by_compatible_status(fdt, DT_STATUS_OK_NSEC | DT_STATUS_OK_SEC, &node, &compat)` ?

Reply:
Ok, today status a bit defined with a single bit hence it will work with `if (node_status & status)`

Reply:
Question is what to differentiate between:
status = okay / secure-status = okay
status = disabled / secure-status = okay
status = okay / secure-status = disabled
I think a straight compare seems the best way for caller to get target status value.

Reply:
status is the value used in linux
secure-status is used in OPTEE
refer to `_fdt_get_status`
If we want to have property only allocated in Secure, we have to ensure that `status` is disabled and `secure-status` is okay

Reply:
Then you can call with `status` set to `DT_STATUS_OK_SEC`.
One may use this function with `DT_STATUS_OK_SEC` or with `DT_STATUS_OK_SEC | DT_STATUS_OK_SEC` for some other means.

By the way: do you really expect several to find several compatible node but one 1 with a non matching status value?

Reply:
Why not? I'm not doing the Linux DTB.

Reply:
> > By the way: do you really expect several to find several compatible node but one 1 with a non matching status value?
>
> Why not? I'm not doing the Linux DTB.

That's fine. I was just curious.

The util function you're implementing can be used by others platforms/drivers. It should be clear what's its purpose is, hence a clear description of what is expected (in value and behavior) for argument `status`.